modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3336,7 +3341,8 @@ void GDScriptAnalyzer::reduce_identifier(GDScriptParser::IdentifierNode *p_ident
 	}
 
 	if (found_source) {
-		if ((p_identifier->source == GDScriptParser::IdentifierNode::MEMBER_VARIABLE || p_identifier->source == GDScriptParser::IdentifierNode::INHERITED_VARIABLE) && parser->current_function && parser->current_function->is_static) {
+		bool source_is_variable_or_signal = p_identifier->source == GDScriptParser::IdentifierNode::MEMBER_VARIABLE || p_identifier->source == GDScriptParser::IdentifierNode::INHERITED_VARIABLE || p_identifier->source == GDScriptParser::IdentifierNode::MEMBER_SIGNAL;
```

Comment:
Since this adding signals, the error message should be chaged to reflect it, otherwise it might be confusing.

Reply:
Fixed the error message!